E/eccodes/build_tarballs.jl

```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "eccodes"
+version = v"2.17.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://confluence.ecmwf.int/download/attachments/45757960/eccodes-2.17.0-Source.tar.gz", "762d6b71993b54f65369d508f88e4c99e27d2c639c57a5978c284c49133cc335")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+sed -e '21,93d' eccodes-2.17.0-Source/cmake/eccodes_test_endiness.cmake > new_endianess_test
+echo 'set( IEEE_BE ${ECCODES_BIG_ENDIAN} )' >> new_endianess_test
+echo 'set( IEEE_LE ${ECCODES_LITTLE_ENDIAN} )' >> new_endianess_test
+mv new_endianess_test eccodes-2.17.0-Source/cmake/eccodes_test_endiness.cmake
+if [ ${target} = "x86_64-w64-mingw32" ] || [ ${target} = "i686-w64-mingw32"] ; then 
+    sed -e '318d' -e '320d' -e '322d' -e '358d' eccodes-2.17.0-Source/CMakeLists.txt -e '26 a add_compile_definitions(ECCODES_ON_WINDOWS)'> new_cmakelist
+    mv new_cmakelist eccodes-2.17.0-Source/CMakeLists.txt
+    sed -e '425d' eccodes-2.17.0-Source/src/grib_context.c > new_context
+    mv new_context eccodes-2.17.0-Source/src/grib_context.c
+    chmod +x eccodes-2.17.0-Source/cmake/ecbuild_windows_replace_symlinks.sh 
+else 
+    sed -e '318d' -e '320d' -e '322d' -e '358d' eccodes-2.17.0-Source/CMakeLists.txt > new_cmakelist
```

Comment:
Just a comment: I find these `sed` scripts very fragile in case you'll want to update the version of the upstream package.  Why not using patches instead?  At least they would be tolerant to line shifts

Reply:
I wanted to use patches, but I couldn't figure out how to get the files into the environment. Is it documented somewhere? What's [in the docs now](https://juliapackaging.github.io/BinaryBuilder.jl/stable/build_tips.html#Editing-files-in-the-wizard-1) looked even more finicky since you have to write the diff by hand in the script.

Reply:
Note that the "stable" version is extremely out-of-date :grimacing:  Look at the "dev" instead: https://juliapackaging.github.io/BinaryBuilder.jl/dev/

Anyway:
* create a subdirectory called `bundled/patches` and put the patch in it
* add
  ```julia
  DirectorySource("./bundled"),
  ```
  to the list of sources
* add this command to the build script to patch the file
  ```sh
  atomic_patch -p1 /workspace/patches/filename.patch
  ```

Reply:
There are plenty of examples of recipes using [`atomic_patch`](https://github.com/JuliaPackaging/Yggdrasil/search?q=atomic_patch&unscoped_q=atomic_patch)